lib/suites/LinkedDataSignature.js

```diff
@@ -11,37 +11,57 @@ const LinkedDataProof = require('./LinkedDataProof');
+   * Either a `key` OR at least one of `signer`/`verifier` is required:
+   *
+   * @param {object} [options.key] - An optional key object (containing an
+   *   `id` property, and either `signer` or `verifier`, depending on the
+   *   intended operation. Useful for when the application is managing keys
+   *   itself (when using a KMS, you never have access to the private key,
+   *   and so should use the `signer` param instead).
+   * @param {Function} [options.signer] - Signer function that returns an
+   *   object with an async sign() method. This is useful when interfacing
+   *   with a KMS (since you don't get access to the private key and its
+   *   `signer()`, the KMS client gives you only the signer function to use).
+   * @param {Function} [options.verifier] - Verifier function that returns
+   *   an object with an async `verify()` method. Useful when working with a
+   *   KMS-provided verifier function.
    *
    * Advanced optional parameters and overrides:
    *
-   * @param [proof] {object} a JSON-LD document with options to use for
-   *   the `proof` node (e.g. any other custom fields can be provided here
-   *   using a context different from security-v2).
-   * @typedef {Object} LDKeyPair
-   * @param {LDKeyPair} LDKeyClass - The crypto-ld key type that this suite
-   *   will use to sign/verify signatures.
-   * @param [date] {string|Date} signing date to use if not passed.
-   * @param {function} [signer] - Signer function.
-   * @param [useNativeCanonize] {boolean} true to use a native canonize
-   *   algorithm.
+   * @param {string} [options.verificationMethod]  A key id URL of the paired
+   *   public key. If not provided, this is set from key.id, signer.id or
+   *   verifier.id.
+   * @param {object} [options.proof] - A JSON-LD document with options to use
+   *   for the `proof` node (e.g. any other custom fields can be provided here
+   *   using a context different from security-v2). If not provided, this is
+   *   constructed during signing.
+   * @param {string|Date} [options.date] - Signing date to use if not passed.
+   * @param {boolean} [options.useNativeCanonize] - Whether to use a native
+   *   canonize algorithm.
    */
   constructor({
-    type, verificationMethod, proof, LDKeyClass, date, signer, useNativeCanonize
+    type, verificationMethod, proof, LDKeyClass, date, key, signer, verifier,
+    useNativeCanonize
   } = {}) {
-    // validate common options
-    if(verificationMethod !== undefined &&
-      typeof verificationMethod !== 'string') {
-      throw new TypeError('"verificationMethod" must be a URL string.');
-    }
     super({type});
-    this.verificationMethod = verificationMethod;
     this.LDKeyClass = LDKeyClass;
-    this.signer = signer;
     this.proof = proof;
-    if(date !== undefined) {
+    const vm = this._validateVerificationMethod({
+      verificationMethod, key, signer, verifier
+    });
```

Comment:
seem weird to return `vm` from a `validate` API which normally reports `{valid: true}`.  Can this pattern be used here or does this helper need to be renamed??

Reply:
Yeah, let's rename the helper. (I'll use Dave's suggestion below.)

Reply:
Renamed.